zipkin-cassandra/src/test/scala/com/twitter/zipkin/cassandra/CassandraSpanStoreFactorySpec.scala

```diff
@@ -46,4 +46,38 @@ class CassandraSpanStoreFactorySpec extends FunSuite with Matchers {
       asList(new InetSocketAddress("1.1.1.1", 9143), new InetSocketAddress("2.2.2.2", 9042))
     )
   }
+
+  test("creatingClusterBuilder with authentication") {
+    TestFactory.nonExitingMain(Array(
+      "-zipkin.store.cassandra.auth", "true",
+      "-zipkin.store.cassandra.user", "user",
+      "-zipkin.store.cassandra.password", "pass"
+    ))
+
+    val builder = TestFactory.createClusterBuilder()
+    val authProvider     = get_privates(builder, "authProvider")
+    val user = get_privates(authProvider, "username")
+    val pass = get_privates(authProvider, "password")
+    user should be("user")
+    pass should be("pass")
+  }
+
+  test("creatingClusterBuilder without authentication") {
+    TestFactory.nonExitingMain(Array(
+      "-zipkin.store.cassandra.auth", "false"
+    ))
+
+    val builder = TestFactory.createClusterBuilder()
+    val authProvider     = get_privates(builder, "authProvider")
+    val user = get_privates(authProvider, "username")
```

Comment:
rename to getField? also no need to declare vals.. `getField(authProvider, "username") should be(empty)`

p.s. I'd be surprised if there was no reflection-based matcher, but I'm often surprised.